src/libtiled/wangset.cpp

```diff
@@ -301,6 +319,132 @@ WangSet::WangSet(Tileset *tileset,
     mCornerColors(cornerColors),
     mUniqueFullWangIdCount(0)
 {
+    Q_ASSERT(mEdgeColors <= 15 && mCornerColors <= 15);
+
+    if (mEdgeColors > 1) {
+        mEdges.reserve(mEdgeColors + 1);
+        mEdges.append(new WangColor());
+        for (int i = 1; i <= mEdgeColors; ++i)
+            mEdges.append(new WangColor(i,
+                                        true,
+                                        QString(),
+                                        defaultWangColors[i - 1],
+                                        -1));
+    }
+
+    if (mCornerColors > 1) {
+        mCorners.reserve(mCornerColors + 1);
+        mCorners.append(new WangColor());
+        for (int i = 1; i <= mCornerColors; ++i)
+            mCorners.append(new WangColor(i,
+                                          false,
+                                          QString(),
+                                          defaultWangColors[i - 1],
+                                          -1));
+    }
+}
+
+void WangSet::setEdgeColors(int n)
+{
+    Q_ASSERT(n <= 15);
+
+    if (n == mEdgeColors)
+        return;
+
+    if (n == 1) {
+        mEdgeColors = 1;
+        mEdges.clear();
+        mEdges.append(new WangColor());
+        return;
+    }
+
+    if (n < mEdgeColors) {
+        while (mEdges.size() != n + 1)
+            mEdges.removeLast();
+    } else {
+        while (mEdges.size() != n + 1)
+            mEdges.append(new WangColor(mEdges.size(),
+                                        true,
+                                        QString(),
+                                        defaultWangColors[mEdges.size() - 1],
+                                        -1));
+    }
+
+    mEdgeColors = n;
+}
+
+void WangSet::setCornerColors(int n)
+{
+    Q_ASSERT(n <= 15);
+
+    if (n == mCornerColors)
+        return;
+
+    if (n == 1) {
+        mCornerColors = 1;
+        mCorners.clear();
+        mCorners.append(new WangColor());
+        return;
+    }
+
+    if (n < mCornerColors) {
+        while (mCorners.size() != n + 1)
+            mCorners.removeLast();
+    } else {
+        while (mCorners.size() != n + 1)
+            mCorners.append(new WangColor(mCorners.size(),
+                                          false,
+                                          QString(),
+                                          defaultWangColors[mCorners.size() - 1],
+                                          -1));
+    }
+
+    mCornerColors = n;
```

Comment:
I have made this change, as well as a couple other name changes:
mEdgeColors -> [REMOVED]
mCornerColors -> [REMOVED]
mEdges -> mEdgeColors
mCorners -> mCornerColors
edgeColors() -> edgeColorCount()
cornerColors() -> cornerColorCount()